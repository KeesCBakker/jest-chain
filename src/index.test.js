import './index';

describe('jest-chain', () => {
  it('chains top level matchers', () => {
    expect(1)
      .toBe(1)
      .toBeGreaterThan(0)
      .toBeLessThan(2);
  });

  it('chains nested level matchers', () => {
    expect(1)
      .toBe(1)
      .not.toBe(0)
      .toBeGreaterThan(0)
      .not.toBeGreaterThan(1)
      .toBeLessThan(2)
      .not.toBeLessThan(1);
  });

  it('chains custom matchers from jest-extended', () => {
    expect(1)
      .toBe(1)
      .toBeGreaterThan(0)
      .toBeLessThan(2)
      .toBePositive()
      .not.toBeNegative();
  });

  describe('fails fast', () => {
    it('throws error from first matcher: toBe when 1 != 2', () => {
      expect(() =>
        expect(1)
          .toBe(2)
          .toBeGreaterThan(1)
          .toBeLessThan(1)
      ).toThrowErrorMatchingSnapshot();
    });

    it('throws error from second matcher: toBeGreaterThan when 1 !> 1', () => {
      expect(() =>
        expect(1)
          .toBe(1)
          .toBeGreaterThan(1)
          .toBeLessThan(1)
      ).toThrowErrorMatchingSnapshot();
    });

    it('throws error from second matcher: toBeLessThan when 1 !< 1', () => {
      expect(() =>
        expect(1)
          .toBe(1)
          .toBeGreaterThan(0)
          .toBeLessThan(1)
      ).toThrowErrorMatchingSnapshot();
    });
  });
});
