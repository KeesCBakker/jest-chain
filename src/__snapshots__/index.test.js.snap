// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-chain fails fast throws error from first matcher: toBe when 1 != 2 1`] = `
"<dim>expect(</><red>received</><dim>).toBe(</><green>expected</><dim>) // Object.is equality</>

Expected value to be:
  <green>2</>
Received:
  <red>1</>"
`;

exports[`jest-chain fails fast throws error from second matcher: toBeGreaterThan when 1 !> 1 1`] = `
"<dim>expect(</><red>received</><dim>).toBeGreaterThan(</><green>expected</><dim>)</>

Expected value to be greater than:
  <green>1</>
Received:
  <red>1</>"
`;

exports[`jest-chain fails fast throws error from second matcher: toBeLessThan when 1 !< 1 1`] = `
"<dim>expect(</><red>received</><dim>).toBeLessThan(</><green>expected</><dim>)</>

Expected value to be less than:
  <green>1</>
Received:
  <red>1</>"
`;
